### ISSUE TITLE
Implement to_sql

### ISSUE THREAD

**Issue by Harshitg** [databricks/koalas#1317]
As a feature request, implementing to_sql capabilities would be very helpful

Thanks

**Comment by Harshitg** [comment]
To add, in the meantime one can always convert the kDF into a sDF and push it to sql. The understanding is correct?

**Comment by HyukjinKwon** [comment]
Yup, you can also leverage [DataFrame.to_spark_io](https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.DataFrame.to_spark_io.html) and [koalas.read_spark_io](https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.read_spark_io.html#databricks-koalas-read-spark-io) to read and write from/to external DBMSs.

You might want to take a look at [databricks.koalas.sql](https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.sql.html).

**Comment by Harshitg** [comment]
DataFrame.to_spark_io is amazing at achieving that. I have a unrelated question, the inserts happen row by row instead of batches, isn't it?

**Comment by HyukjinKwon** [comment]
Yes, in general.

**Comment by raxit65535** [comment]
> To add, in the meantime one can always convert the kDF into a sDF and push it to sql. The understanding is correct?

If we convert kDF to sDF, would it add any latency ?



> Yup, you can also leverage [DataFrame.to_spark_io](https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.DataFrame.to_spark_io.html) and [koalas.read_spark_io](https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.read_spark_io.html#databricks-koalas-read-spark-io) to read and write from/to external DBMSs.

> 
> You might want to take a look at [databricks.koalas.sql](https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.sql.html) too.

If we use DataFrame.to_spark_io, what would be the value in `path`.. If there any practical example for using `DataFrame.to_spark_io` with any RDBMS ?

Thanks

**Comment by HyukjinKwon** [comment]
They are virtually same except that there's a bit of overhead to handle metadata which is arguably trivial compared to execution and runtime.

`path` will have to be a URL. e.g.) `hdfs://a/b/c`.

There are examples in the documentation, see also https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.DataFrame.to_spark_io.html.

You could leverage JDBC Spark source via `DataFrame.to_spark_io`, see also https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html
